cmd/catalyst-node/catalyst-node.go

```diff
@@ -462,35 +472,51 @@ func protocol(r *http.Request) string {
 	return "http"
 }
 
-func queryMistForClosestNode(playbackID, lat, lon string) (string, error) {
-	url := fmt.Sprintf("http://localhost:%s/%s", mistUtilLoadPort, playbackID)
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return "", err
-	}
-	if lat != "" && lon != "" {
-		req.Header.Set("X-Latitude", lat)
-		req.Header.Set("X-Longitude", lon)
-	} else {
-		glog.Warningf("Incoming request missing X-Latitude/X-Longitude, response will not be geolocated")
-	}
-	client := &http.Client{}
-	resp, err := client.Do(req)
-	if err != nil {
-		return "", err
-	}
-	defer resp.Body.Close()
+func queryMistForClosestNode(playbackID, lat, lon string, redirectPrefixes []string) (string, error) {
```

Comment:
I think you could improve this part by doing the following:
1. Change this function to `queryMistForClosestNode(playbackID, lat, lon string, redirectPrefix string) (string, error)`
2. Move the for loop outside this function `for _, prefix := range redirectPrefixes`
3. Then, you could add the test for the logic you added.

What do you think?

Reply:
yeah, that is what i should have done in the first place. wrote most of the code while in zombie mode yesterday. updated now :+1: